.github/workflows/test.yml

```diff
@@ -1,47 +1,86 @@
-name: Test TPS
-on:
-- push
-- pull_request
-jobs:
-  pcsf:
-    name: Test PCSF
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: [ubuntu-latest]
-    steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
-      - name: Install conda environment
-        uses: conda-incubator/setup-miniconda@v2
-        with:
-          activate-environment: pcsf
-          environment-file: pcsf/environment.yml
-          auto-activate-base: false
-          auto-update-conda: false
-          miniconda-version: 'latest'
-      # Run the PCSF tests in the pcsf conda environment
-      - name: Test PCSF
-        shell: bash --login {0}
-        run: pytest pcsf/tests/
-  workflow:
-    name: Test TPS workflow
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: [ubuntu-latest]
-    steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
-      - name: Install conda environment
-        uses: conda-incubator/setup-miniconda@v2
-        with:
-          activate-environment: tps_workflow
-          environment-file: workflow/minimal_env.yml
-          auto-activate-base: false
-          miniconda-version: 'latest'
-      # Run the workflow tests
-      - name: Test workflow
-        shell: bash --login {0}
-        # Verbose output and disable stdout and stderr capturing
-        run: pytest -vs workflow/tests/
+name: Test TPS
+on:
+- push
+- pull_request
+jobs:
+  pcsf:
+    name: Test PCSF
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Install conda environment
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          activate-environment: pcsf
+          environment-file: pcsf/environment.yml
+          auto-activate-base: false
+          auto-update-conda: false
+          miniconda-version: 'latest'
+      # Run the PCSF tests in the pcsf conda environment
+      - name: Test PCSF
+        shell: bash --login {0}
+        run: pytest pcsf/tests/
+  workflow:
+    name: Test TPS workflow
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Build TPS Docker image

```

Comment:
Can we skip this step when testing the workflow?  I expect that when using the container, if it isn't already present on the GitHub server it will be pulled from DockerHub.  Pulling is probably faster than building.